File: week13/algorithm1.js
function insertionsort(unsorted) {

    if(unsorted.length === 0) {
        return [];
    }

    let pivot = unsorted[0]
    let left = []
    let right = []
    
    for(let i = 1; i < unsorted.length; i++) {
        if(unsorted[i] <= pivot) {
            left.push(unsorted[i]);
        } else {
            right.push(unsorted[i]);
        }
    }

    return insertionsort(left).concat([pivot].concat(insertionsort(right)));
}

b = []

for(let i = 0; i < 10; i++) {
    a = Math.round(Math.random() * 10)
    b.push(a)
}

console.log(b)
console.log(insertionsort(b))

File: week13/algorithm1.py
from random import (
    randrange, 
    seed,
)

def insertionsort(unsorted):

    if len(unsorted) == 0:
        return []
    
    pivot = unsorted[0]
    rest = unsorted[1:]

    left = list()
    right = list()

    for x in rest:
        if x <= pivot:
            left.append(x)
        else:
            right.append(x)

    return insertionsort(left) + [pivot] + insertionsort(right)


if __name__ == "__main__":
    seed(42)
    a = [randrange(10) for x in range(10)]

    print(a)
    print(insertionsort(a))